# Covid 19 UK Deaths
In this Jupyter notebook, I will be conducting an exploratory data analysis (EDA) on three datasets related to the COVID-19 pandemic in the United Kingdom. The datasets I have chosen are Deaths.csv, Vaccinations.csv, and ConfirmedCasesByDateReported.csv. It's important to note that the 'areaName' column in ConfirmedCasesByDateReported.csv and Deaths.csv only contain 'United Kingdom'. The 'areaName' column in Vaccinations.csv, only contain information for the four countries that make up the United Kingdom. Throughout this analysis, I will be using pandas to manipulate and explore the datasets, and I will also be using visualizations to better understand the data..



In [79]:
import numpy as np
import pandas as pd
pd.options.mode.chained_assignment = None  # default='warn'
import matplotlib.pyplot as plt
import seaborn as sns

Creating Dataframes

In [80]:
d = pd.read_csv("data/Deaths.csv") # creates data frame
v = pd.read_csv("data/Vaccinations.csv")
c = pd.read_csv("data/ConfirmedCasesByDateReported.csv")

In [81]:
d.head()

areaName       date  newDeaths28DaysByDeathDate  \
0  United Kingdom  10/3/2021                          32   
1  United Kingdom   9/3/2021                          97   
2  United Kingdom   8/3/2021                         130   
3  United Kingdom   7/3/2021                         102   
4  United Kingdom   6/3/2021                         128   

   cumDeaths28DaysByDeathDate  
0                    125165.0  
1                    125133.0  
2                    125036.0  
3                    124906.0  
4                    124804.0

In [82]:
d.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 376 entries, 0 to 375
Data columns (total 4 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   areaName                    376 non-null    object 
 1   date                        376 non-null    object 
 2   newDeaths28DaysByDeathDate  376 non-null    int64  
 3   cumDeaths28DaysByDeathDate  374 non-null    float64
dtypes: float64(1), int64(1), object(2)
memory usage: 11.9+ KB


There are two missing values in the 'cumDeaths28DaysByDeathDate' column, as the non-null count for this column is 374 instead of 376. It may be necessary to handle these missing values before proceeding with any further analysis.

In [83]:
d.describe()

newDeaths28DaysByDeathDate  cumDeaths28DaysByDeathDate
count                  376.000000                  374.000000
mean                   332.885638                51317.727273
std                    344.537116                32077.730134
min                      0.000000                    1.000000
25%                     31.750000                38706.750000
50%                    221.500000                41679.500000
75%                    487.750000                63238.500000
max                   1359.000000               125165.000000

In [84]:
# Drop the two rows with null values.
d.dropna(inplace=True)

In [85]:
v.head()

areaName       date  weeklyPeopleVaccinatedFirstDoseByVaccinationDate  \
0  England  14/2/2021                                           2270123   
1  England   7/2/2021                                           2398160   
2  England  31/1/2021                                           2198114   
3  England  24/1/2021                                           2244887   
4  England  17/1/2021                                           1608244   

   cumPeopleVaccinatedFirstDoseByVaccinationDate  \
0                                       12844193   
1                                       10574070   
2                                        8175910   
3                                        5977796   
4                                        3732909   

   weeklyPeopleVaccinatedSecondDoseByVaccinationDate  \
0                                              17521   
1                                               8571   
2                                               8466   
3                                               6527   
4                                              40483   

   cumPeopleVaccinatedSecondDoseByVaccinationDate  
0                                          487697  
1                                          470176  
2                                          461605  
3                                          453139  
4                                          446612

In [86]:
v.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40 entries, 0 to 39
Data columns (total 6 columns):
 #   Column                                             Non-Null Count  Dtype 
---  ------                                             --------------  ----- 
 0   areaName                                           40 non-null     object
 1   date                                               40 non-null     object
 2   weeklyPeopleVaccinatedFirstDoseByVaccinationDate   40 non-null     int64 
 3   cumPeopleVaccinatedFirstDoseByVaccinationDate      40 non-null     int64 
 4   weeklyPeopleVaccinatedSecondDoseByVaccinationDate  40 non-null     int64 
 5   cumPeopleVaccinatedSecondDoseByVaccinationDate     40 non-null     int64 
dtypes: int64(4), object(2)
memory usage: 2.0+ KB


In this case, there are no null values.

In [87]:
c.head()

areaName       date  newCasesByPublishDate
0  United Kingdom  25/3/2021                   6397
1  United Kingdom  24/3/2021                   5605
2  United Kingdom  23/3/2021                   5379
3  United Kingdom  22/3/2021                   5342
4  United Kingdom  21/3/2021                   5312

In [88]:
c.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 421 entries, 0 to 420
Data columns (total 3 columns):
 #   Column                 Non-Null Count  Dtype 
---  ------                 --------------  ----- 
 0   areaName               421 non-null    object
 1   date                   421 non-null    object
 2   newCasesByPublishDate  421 non-null    int64 
dtypes: int64(1), object(2)
memory usage: 10.0+ KB


As with vaccinations, there are no null values here.